### PR TITLE
default to api_key, fallback on username/password

### PR DESCRIPTION
**Pull request by seanstory**
## Closes https://github.com/elastic/connectors-python/issues/1468

When I moved default configs to the code instead of the YAML, I added logic that merges the default config dict with the config-file provided config dict. This introduced an edge case where the default provided username/password auth for Elasticsearch, but the user provided API key auth for Elasticsearch, and we raised an error.

However - there's really no reason for this to be an error state. If both are configured, we'll just default to using API Key. That's the one we want folks using.

Alternatively, we _could_ put effort into trying to reconcile the two configurations, and discount certain default configs if other explicit configs are provided. But I'm not sure this is worth the effort.

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [ ] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

**Review by wangch079 — COMMENTED** [review]
did the same thing here: https://github.com/elastic/connectors-python/pull/1262 😆 

Worth adding some tests

File: connectors/es/client.py [hunk 1]
```diff
@@ -50,17 +50,13 @@ def __init__(self, config):
         }
         logger.debug(f"Host is {self.host}")
 
-        if "username" in config:
```
Comment (timgrein) [inline connectors/es/client.py:53]: We could also add a warning log that we found both in the config and use the `api_key`, so someone could trace this behavior during support/debugging, if needed.

**Review by timgrein — COMMENTED** [review]
++ on adding a test here (and maybe a warning log), this error looks familiar as @wangch079 said

File: connectors/es/client.py (hunk 1 above)
Comment (seanstory) [reply connectors/es/client.py:53]: warn log is a good idea

**Review by artem-shelkovnikov — APPROVED** [review]
Good stuff!

**Review by artem-shelkovnikov — APPROVED** [review]

**Pull request merged**